OMCompiler/SimulationRuntime/c/util/utility.c

```diff
@@ -160,32 +160,12 @@ void OpenModelica_updateUriMapping(threadData_t *threadData, void *namesAndDirs)
 
 static const char *PATH_NOT_IN_FMU_RESOURCES = "Returning path (%s) not in the resources directory. The FMU might not work as expected if you send it to a different system";
 
-static int isWindows(void)
+/* if uri starts with X:\ or X:/ */
+static int hasDriveLetter(const char* uri)
 {
-#if defined(__MINGW32__) || defined(_MSC_VER)
-  return 1;
-#else /* not windows */
-  return 0;
-#endif
-}
-
-static int isMinGW(void)
-{
-#if defined(__MINGW32__)
-  return 1;
-#else /* not MinGW */
-  return 0;
-#endif
-}
-
-static int isDriveLetter(char c)
-{
-  return isWindows() && ((c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z'));
-}
-
-static int isPathSeparator(char c)
-{
-  return (c == '/') || (isWindows() && (c == '\\'));
+  return strlen(uri) > 2 &&
+         ((uri[0] >= 'A' && uri[0] <= 'Z') || (uri[0] >= 'a' && uri[0] <= 'z')) &&
```

Comment:
Maybe have uri[0] >= 'A' && uri[0] <= 'Z' and the other clauses as named variables

Reply:
yeah, that would save on indexing, but this operation only happens once when we instantiate the fmu so it doesn't matter that much